agent/api/ecsclient/client.go

```diff
@@ -292,19 +292,78 @@ func (client *APIECSClient) SubmitTaskStateChange(change api.TaskStateChange) er
 	}
 
 	status := change.Status.String()
-	_, err := client.submitStateChangeClient.SubmitTaskStateChange(&ecs.SubmitTaskStateChangeInput{
-		Cluster: &client.config.Cluster,
-		Task:    &change.TaskArn,
-		Status:  &status,
-		Reason:  &change.Reason,
-	})
+
+	req := ecs.SubmitTaskStateChangeInput{
+		Cluster: aws.String(client.config.Cluster),
+		Task:    aws.String(change.TaskArn),
+		Status:  aws.String(status),
+		Reason:  aws.String(change.Reason),
+	}
+
+	containerEvents := make([]*ecs.ContainerStateChange, len(change.Containers))
+	for i, containerEvent := range change.Containers {
+		containerEvents[i] = client.buildContainerStateChangePayload(containerEvent)
+	}
+
+	req.Containers = containerEvents
+
+	_, err := client.submitStateChangeClient.SubmitTaskStateChange(&req)
 	if err != nil {
 		log.Warn("Could not submit a task state change", "err", err)
 		return err
 	}
+
+	// WIP LOG payload
+	seelog.Info("WIP, sent req contents: ", req)
+
 	return nil
 }
 
+func (client *APIECSClient) buildContainerStateChangePayload(change api.ContainerStateChange) *ecs.ContainerStateChange {
+	statechange := &ecs.ContainerStateChange{
+		ContainerName: aws.String(change.ContainerName),
+	}
+
+	if change.Reason != "" {
```

Comment:
so `change api.ContainerStateChange` is a struct value and [can't be nil](https://play.golang.org/p/IgdG6wvFjr). at worst it may have [its fields zeroed if no value is specified](https://golang.org/ref/spec#The_zero_value).